File: CS/DXPivotGrid_CustomFilterPopup/CustomFilterPopupTemplateSelector.cs
using DevExpress.Xpf.PivotGrid.Internal;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;

namespace DXPivotGrid_CustomFilterPopup {
    class CustomFilterPopupTemplateSelector : FilterPopupTemplateSelector {
        public DataTemplate CustomTemplate { get; set; }
        public override DataTemplate SelectTemplate(object item, DependencyObject container) {
        var field = FieldHeader.GetHeader(container)?.Field;
            if(field != null && field.Group == null)
                return CustomTemplate;
            return LegacyTemplate;
        }
    }
}

File: VB/DXPivotGrid_CustomFilterPopup/CustomFilterPopupTemplateSelector.vb
Imports DevExpress.Xpf.PivotGrid.Internal
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows

Namespace DXPivotGrid_CustomFilterPopup
	Friend Class CustomFilterPopupTemplateSelector
		Inherits FilterPopupTemplateSelector

		Public Property CustomTemplate() As DataTemplate
		Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
		Dim field = FieldHeader.GetHeader(container)?.Field
			If field IsNot Nothing AndAlso field.Group Is Nothing Then
				Return CustomTemplate
			End If
			Return LegacyTemplate
		End Function
	End Class
End Namespace
